clients/src/main/java/org/apache/kafka/common/telemetry/metrics/SinglePointMetric.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.metrics;
+
+/**
+ * This class represents a metric that does not yet contain resource tags.
+ * These additional resource tags will be added before emitting metrics by the telemetry reporter.
+ */
+public class SinglePointMetric implements MetricKeyable {
```

Comment:
`internals` package?

Reply:
The only thing which is not clear to me in the `org.apache.kafka.common` module is the true definition of `internals` i.e. `org.apache.kafka.common.internals` hold handful of classes which seems for internal usage but other packages do not have any differentiation. For example `network`, which consists of classes like `SslTransportLayer`, etc. are these not internal? Similarly for other packages. It might be very naive question but certainly I am not able to get the true definition here.

@kirktrue @philipnee Please if I can help me with the query.